Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -316,6 +320,13 @@ WarpXOpenPMDPlot::Init (openPMD::Access access, bool isBTD)
         m_MPIRank = 1;
     }
 
+    if ( m_OpenPMDFileType.compare("bp") == 0 )
```

Comment:
I think we should make this a user input option.
We can default to `variableBased`, fallback default: `groupBased`, but the user might want to pick `fileBased` for various reasons

Reply:
Sure, so  in addition to parameter "openpmd_backend",  I will introduce another one called "openpmd_iteration". It is default to variableBased if available, otherwise, group based.

Reply:
actually, I will rename openpmd_tspf to openpmd_iteration

Reply:
You could encode it with an enum so that the user can pick between all three encodings :)